doc: annotate all ABI fields added/changed in 3.17 ABI

kv2019i opened the pull request:
Add a "ABImajor.minor" note to inline documentation of all
ABI interfaces that have been added or modified as part of
3.17 interface changes.

This is patch serves as an example for proposed new policy on this topic, making the annotations mandatory for any new change:
https://github.com/thesofproject/sof-docs/pull/283

lgirdwood reviewed (approved):
Thanks @kv2019i - this is also a reminder for me to check this in PRs.

marc-hb commented:
Any reason not to convert modified comments to doxygen syntax too?

lgirdwood commented:
@kv2019i ping

kv2019i commented:
@marc-hb wrote:
> Any reason not to convert modified comments to doxygen syntax too?

Not really. I don't see we actually generate doxygen output from these and thus I just used existing conventions of using 

```
  /* simple comment when before a field */
  struct foobar;

  int barfield; /**< doxygen style when after the field */
```

But yeah, why existings docs are like this, I don't really know.

kv2019i commented:
@marc-hb Now updated. I converted all comments to doxygen style and verified from doxygen output that the documentation actually ends up there. It's not included in our public website docs, but at least it's there if someone generates the local documentation.
